bundler/lib/bundler/fetcher.rb

```diff
@@ -102,11 +102,11 @@ def fetch_spec(spec)
       uri = Bundler::URI.parse("#{remote_uri}#{Gem::MARSHAL_SPEC_DIR}#{spec_file_name}.rz")
       if uri.scheme == "file"
         path = Bundler.rubygems.correct_for_windows_path(uri.path)
-        Bundler.load_marshal Bundler.rubygems.inflate(Gem.read_binary(path))
+        Bundler.safe_load_marshal Bundler.rubygems.inflate(Gem.read_binary(path))
```

Comment:
let's add a test that has an exception raised when attempting to load an unexpected class through this codepath? and same with the one in `RubyGemsIntegration.fetch_specs`